### PR TITLE
When rotating, don't retry 403s

### PR DESCRIPTION
**Pull request by seamusabshere**
Otherwise we get tons of log spam, etc.

### CODE REVIEW

File: prepare.ts [hunk 1]
```diff
@@ -149,6 +149,12 @@ export default function prepare(
...
       },
     });
+    if (response.statusCode == 403) {
+      console.error(
+        "vault responded with 403 access denied when i tried to rotate, giving up"
+      );
+      return;
+    }
     if (sync) {
       onLease(vaultPath, parseLeaseResponse(response));
```
Comment (emk) [inline prepare.ts:159]: Does `onLease` re-install the timer on the happy path? Once you print the error, we should make sure that the cred rolling callback stops running, either by not reinstalling it, or by stopping it if it's a periodic timer.

**Review by emk — COMMENTED** [review]
A question...

File: prepare.ts (hunk 1 above)
Comment (seamusabshere) [reply prepare.ts:159]: @emk do you mean stopping all future invocations of onLease?

File: prepare.ts [hunk 2]
```diff
@@ -141,33 +141,59 @@ export default function prepare(
...
+          "X-Vault-Token": VAULT_TOKEN,
+        },
+      });
+      try {
+        onLease(vaultPath, parseLeaseResponse(checkStatusCode(response)));
+      } catch (e) {
+        console.error(e.message);
+        if (!(e instanceof RetryAuthFailure)) {
+          throw e;
+        }
```
Comment (CharlieMcVicker) [inline prepare.ts:170]: It feels a little funky to 1) expect an exception where we could use branching and 2) to expect something inline to throw that error. Let me see if I can write it branched and compare. I'll open a pr and link here.

**Comment by CharlieMcVicker** [comment]
Right now, what happens when we do get this new error? It looks like we never throw it so this function will just be a no-op. Is that on purpose?

**Pull request merged**